### PR TITLE
Add logging to ExceptionHandler

### PR DESCRIPTION
**Pull request by sctape**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
         "equip/structure": "^1.0",
         "filp/whoops": "^2.0",
         "lukasoppermann/http-status": "^2",
+        "monolog/monolog": "^1.19",
```
Comment (shadowhand) [inline composer.json]: We shouldn't require monolog specifically, but rather [`psr/log`](https://packagist.org/packages/psr/log).

Comment (shadowhand) [inline composer.json]: `monolog/monolog` should be recommended.

Comment (sctape) [inline composer.json]: Don't we need to provide an implementation of PSR-3? Like is done with [DiactorosConfiguration](https://github.com/equip/framework/blob/master/src/Configuration/DiactorosConfiguration.php).

Comment (sctape) [inline composer.json]: My concern is that if we don't provide a default configuration that includes an implementation of PSR-3, things will blow up when trying to handle exceptions out of the box.

Comment (shadowhand) [inline composer.json]: Logging should be optional.

Comment (ameech) [inline composer.json]: Why? I can see us having an option to disable logging, but seems like a pretty important piece to have.

Comment (shadowhand) [inline composer.json]: The point of PSR-3 is that you don't need a specific logger. If anything, we should require monolog via equip/project. But not here.

Comment (sctape) [inline composer.json]: @shadowhand can you think of a better way to log exceptions in `ExceptionHandler` without requiring an implementation of `LoggerInterface`?
```
try {
     $this->injector->make(LoggerInterface::class)->error($e->getMessage(), [
         'exception' => $e
     ]);
} catch (InjectionException $e){

}
```

Comment (shadowhand) [inline composer.json]: You can't access the injector in the exception handler, that would be a service locator and violate inversion.

Comment (shadowhand) [inline composer.json]: I was thinking something along the lines of:

```php
public function __construct(
    ...,
    LoggerInterface = null
) { ... }
```

Along with

```php
if ($this->logger) {
    $this->logger->error(...);
}
```

In the handler itself.

Comment (sctape) [inline composer.json]: @shadowhand Gotcha. I'll do that. I didn't look closely enough at how dependencies are [resolved](https://github.com/rdlowrey/auryn#dependency-resolution).

File: composer.json [hunk 2]
```diff
@@ -26,11 +27,13 @@
     "require-dev": {
         "josegonzalez/dotenv": "^1.2",
         "league/plates": "^3.1",
+        "monolog/monolog": "^1.19",
```
Comment (sctape) [inline composer.json]: I added this here following the behavior of how Plates is configured in Equip

File: src/Handler/ExceptionHandler.php [hunk 3]
```diff
@@ -34,18 +36,26 @@ class ExceptionHandler
...
      * @param Whoops $whoops
+     * @param LoggerInterface|null $logger
      */
     public function __construct(
         ExceptionHandlerPreferences $preferences,
         Negotiator $negotiator,
         ResolverInterface $resolver,
-        Whoops $whoops
+        Whoops $whoops,
+        LoggerInterface $logger = null
```
Comment (shadowhand) [inline src/Handler/ExceptionHandler.php]: This is good because it works with existing configurations, so we can include it in a minor release.

**Comment by shadowhand** [comment]
:+1:

**Comment by ameech** [comment]
Nice! 🎉

**Pull request merged**